crates/forge/src/lib.rs

```diff
@@ -110,7 +109,7 @@ pub async fn run(
     let test_crates = test_crates
         .into_iter()
         .map(|tc| tests_filter.filter_tests(tc))
-        .collect_vec();
+        .collect::<Result<Vec<CompiledTestCrateRaw>>>()?;
```

Comment:
Do we need this specialization?

Reply:
![Zrzut ekranu 2023-11-21 o 16 07 10](https://github.com/foundry-rs/starknet-foundry/assets/15279784/9985daf2-226f-4be8-8790-c9911958137e)
I changed the `tests_filter` type